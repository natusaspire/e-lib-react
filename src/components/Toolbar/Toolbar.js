import React, { memo } from 'react';

const Toolbar = ({ title, children }) => (
  <div className="toolbar">
    <div className="toolbar__body">
      <input type="checkbox" id="toolbar-toggle" />
      <label htmlFor="toolbar-toggle">
        <div className="toolbar__title">
          <span>{title}</span>
          <i className="material-icons">&#xE313;</i>
        </div>
      </label>
      <div className="toolbar__dropdown">{children}</div>
    </div>
  </div>
);

export default memo(Toolbar);
