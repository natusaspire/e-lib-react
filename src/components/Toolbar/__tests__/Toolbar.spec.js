import React from 'react';
import { shallow } from 'enzyme';

import Toolbar from '@/components/Toolbar';

describe('<Toolbar />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Toolbar title="Toolbar">
        <div>Tools</div>
      </Toolbar>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
