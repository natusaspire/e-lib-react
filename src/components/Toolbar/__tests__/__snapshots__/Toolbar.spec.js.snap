// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar /> should render correctly 1`] = `
<div
  className="toolbar"
>
  <div
    className="toolbar__body"
  >
    <input
      id="toolbar-toggle"
      type="checkbox"
    />
    <label
      htmlFor="toolbar-toggle"
    >
      <div
        className="toolbar__title"
      >
        <span>
          Toolbar
        </span>
        <i
          className="material-icons"
        >
          îŒ“
        </i>
      </div>
    </label>
    <div
      className="toolbar__dropdown"
    >
      <div>
        Tools
      </div>
    </div>
  </div>
</div>
`;
